pkgs/by-name/ad/add-determinism/package.nix

```diff
@@ -0,0 +1,67 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  zlib,
+  python3,
+  stdenv,
+}:
+
+let
+  marshalparser = python3.pkgs.callPackage ./marshalparser.nix { };
+  pyEnv = python3.withPackages (ps: [ marshalparser ]);
+  cargoLockFile = ./Cargo.lock;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "add-determinism";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "keszybz";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-YfY0l57SjBfO5nLlCfEBzwyVPdzZKK5YKVvVATsObi0=";
+  };
+
+  # this project has no Cargo.lock now
+  cargoLock = {
+    lockFile = cargoLockFile;
+  };
+
+  patches = [
+    ./fix-darwin.patch
+  ];
+
+  postPatch = ''
+    cp ${cargoLockFile} Cargo.lock
+    chmod u+w Cargo.lock
+  '';
+
+  doCheck = !stdenv.hostPlatform.isDarwin; # it seems to be running forever on darwin
+
+  passthru = { inherit pyEnv marshalparser; };
+
+  nativeBuildInputs = [
+    pyEnv
+    pkg-config
+  ];
+
+  propagatedBuildInputs = [ pyEnv ];
```

Comment:
We should wrap the resulting binary instead, otherwise it won't work outside of stdenv.